pkg/rpm/resolver/resolver.go

```diff
@@ -0,0 +1,209 @@
+		}
+
+		rpmNames, err := rpm.CopyRPMs(r.customRPMDir, dest)
+		if err != nil {
+			return fmt.Errorf("copying local rpms to %s: %w", dest, err)
+		}
+		r.rpms = rpmNames
+	}
+
+	if err := r.writeDockerfile(); err != nil {
+		return fmt.Errorf("writing dockerfile: %w", err)
+	}
+
+	zap.L().Info("Resolver image context setup successful")
+	return nil
+}
+
+func (r *Resolver) writeDockerfile() error {
+	values := struct {
+		BaseImage   string
+		RegCode     string
+		AddRepo     string
+		CacheDir    string
+		PkgList     string
+		FromRPMPath string
+		ToRPMPath   string
+	}{
+		BaseImage: baseImageRef,
+		RegCode:   r.packages.RegCode,
+		AddRepo:   strings.Join(r.packages.AdditionalRepos, " "),
+		CacheDir:  r.generateRPMRepoPath(),
+		PkgList:   strings.Join(r.getPKGForResolve(), " "),
+	}
+
+	if r.customRPMDir != "" {
+		values.FromRPMPath = filepath.Base(r.generateRPMPathInBuildContext())
+		values.ToRPMPath = r.generateLocalRPMDirPath()
+	}
+
+	data, err := template.Parse(dockerfileName, dockerfileTemplate, &values)
+	if err != nil {
+		return fmt.Errorf("parsing %s template: %w", dockerfileName, err)
+	}
+
+	filename := filepath.Join(r.generateBuildContextPath(), dockerfileName)
+	if err = os.WriteFile(filename, []byte(data), fileio.ExecutablePerms); err != nil {
+		return fmt.Errorf("writing prepare base image script %s: %w", filename, err)
+	}
+
+	return nil
+}
+
+func (r *Resolver) getPKGForResolve() []string {
+	list := []string{}
+
+	if len(r.packages.PKGList) > 0 {
+		list = append(list, r.packages.PKGList...)
+	}
+
+	if len(r.rpms) > 0 {
+		// generate RPM paths as seen in the resolver image,
+		// needed so that 'zypper install' can locate the rpms
+		for _, name := range r.rpms {
+			list = append(list, filepath.Join(r.generateLocalRPMDirPath(), name))
+		}
+	}
+	return list
+}
+
+func (r *Resolver) getPKGInstallList() []string {
+	list := []string{}
+
+	if len(r.packages.PKGList) > 0 {
+		list = append(list, r.packages.PKGList...)
+	}
+
+	if len(r.rpms) > 0 {
+		// generate the RPMs as package names,
+		// so that zypper can locate them in the RPM repository
+		for _, name := range r.rpms {
+			list = append(list, strings.TrimSuffix(name, filepath.Ext(name)))
+		}
+	}
+	return list
+}
+
+// path to the build dir, as seen in the EIB image
+func (r *Resolver) generateBuildContextPath() string {
+	return filepath.Join(r.dir, "build")
+}
+
+// path to the rpms directory in the resolver build context, as seen in the EIB image
+func (r *Resolver) generateRPMPathInBuildContext() string {
+	return filepath.Join(r.generateBuildContextPath(), "rpms")
+}
+
+// path to rpm cache dir, as seen in the resolver image
+func (r *Resolver) generateRPMRepoPath() string {
+	return filepath.Join(os.TempDir(), rpmRepoName)
+}
+
+// path to the dir containing local rpms, as seen in the resolver image
+func (r *Resolver) generateLocalRPMDirPath() string {
+	return filepath.Join(r.generateRPMRepoPath(), "local")
```

Comment:
Yes, at this directory we copy all user provided RPMs to the resolver image. This is needed so that the [zypper install](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/rpm/resolver/templates/Dockerfile.tpl#L30) command could have access to this RPMs and could successfully resolve their package dependencies. The RPMs are also directly copied to the resolved RPM directory which is returned by the resolver logic.